frame/executive/src/lib.rs

```diff
@@ -201,6 +229,25 @@ where
 		);
 	}
 
+	/// Returns if the runtime was upgraded since the last time this function was called.
```

Comment:
Not sure if it is intentional or not, but this function will return `true` (IIUC) the first time it was called. If that's intentional worth to call it out?

Reply:
So basically, when we run a brand new node, if there is any `on_runtime_upgrade`, it will happen. That seems like a potential issue. We should add into the genesis config generation the correct value then?

Reply:
It will run on genesis. What should be the potential issue? If some upgrade code would run out of nowhere, it means that this upgrade code is already not implemented properly.

Reply:
I don't think it's especially expected for it to run on genesis. Once Kusama is upgraded to include this change, I would default to not running if there's not already a version stored.

Reply:
Yeah, it would run for the first block.

How should this work for other chains? We somehow need to store the version to know what was the last time we have upgraded the runtime.

I don't see the problem of running it for the first block. If there would be code that runs some weird migrations, it would mean that this would happen at the next upgrade anyway. Upgrade code should be protected in such a way that it can not be executed twice and especially it should not be executed on a fresh genesis.

Reply:
The function is called `on_runtime_upgrade`. It's common sense that it shouldn't execute unless a runtime upgrade occured, which is not the case at genesis.

Reply:
We now don't run on genesis.